dirty_cat/_fuzzy_join.py

```diff
@@ -7,31 +7,155 @@
   neighbor and establish a match.
   3. We match the tables using the previous information.
 Categories from the two tables that share many sub-strings (n-grams)
-have greater probability of beeing matched together. The join is based on
+have greater probability of being matched together. The join is based on
 morphological similarities between strings.
+Joining on numerical columns is also possible based on the Euclidean distance.
 """
 
 import warnings
-from typing import Literal, Tuple, Union
+from collections.abc import Iterable
+from typing import List, Literal, Optional, Tuple, Union
 
 import numpy as np
 import pandas as pd
-from scipy.sparse import vstack
+from scipy.sparse import hstack, vstack
 from sklearn.feature_extraction.text import (
     HashingVectorizer,
     TfidfTransformer,
     _VectorizerMixin,
 )
 from sklearn.neighbors import NearestNeighbors
+from sklearn.preprocessing import StandardScaler
+
+
+def _numeric_encoding(main, main_cols, aux, aux_cols):
+    """Encoding numerical columns.
+
+    Parameters
+    ----------
+    main : pd.DataFrame
+        A table with numerical columns.
+    main_cols : str or list
+        The columns of the main table.
+    aux : pd.DataFrame
+        Another table with numerical columns.
+    aux_cols : str or list
+        The columns of the aux table.
+
+    Returns
+    -------
+    main_array : array-like
+        An array of the encoded columns of the main table.
+    aux_array : array-like
+        An array of the encoded columns of the aux table.
+    """
+    aux_array = aux[aux_cols].to_numpy()
+    main_array = main[main_cols].to_numpy()
+    # Re-weighting to avoid measure specificity
+    scaler = StandardScaler()
+    scaler.fit(np.concatenate((aux_array, main_array)))
```

Comment:
Nitpick: Since we use `vstack` in `_string_encoding`, let's use it here as well since it's a bit more explicit than `concatenate`.
```suggestion
    scaler.fit(np.vstack((aux_array, main_array)))
```